content/guide/ios-runtime-types.md

```diff
@@ -0,0 +1,193 @@
+---
+title: iOS Runtime Types
+---
+
+NativeScript provides the `interop` module for working with native C types, pointers, pointer arithmetic and memory.
+
+
+<!-- TODO: Add examples -->
+
+## Using interop
+
+### Creating a pointer
+
+The `Pointer` type allows you to represent a void*.
+
+#### Using the constructor
+
+To create a new pointer with the given offset, use the construcotr:
+
+```ts
+   const interop1 = new interop.Pointer(34)
+```
+
+#### Creating a pointer by adding an offeset from an existing pointer
+
+ ```ts
+const interop2 = interop1.add(4)
+ ```
+
+#### Creating a pointer by removing an offeset to an existing pointer
+
+ ```ts
+const interop3 = interop1.subtract(3)
+ ```
+
+## interop API
+
+### Pointer.toNumber()
+
+```ts
+
+pointerInstance.toNumber()
+
+```
+
+Converts the value of a `Pointer` instance to a number.
+
+---
+
+### adopt
+
+```ts
+
+interop.adopt(ptr: Pointer): AdoptedPointer
+
+```
+
+Makes the specified pointer adopted.
+
+---
+
+### free
+
+```ts
+
+interop.free(ptr: Pointer): void
+
+```
+
+Releases the memory of the specified unadopted pointer.
+
+---
+
+### sizeof
+
+```ts
+
+interop.sizeof(type: any): number
+
+```
+Returns the size of the provided type. The `type` can be: a class constructor (of Objective-C interface), an instance (wrapper of Objective-C interface), struct constructor, struct instance, reference, protocol, function (for c function), fundamental types
+
+---
+
+### alloc
+
+```ts
+
+interop.alloc(size: number): AdoptedPointer
+
+```
+
+Allocates memory `size` in bytes.
+
+---
+
+### handleof
+
+```ts
+
+interop.handleof(instance: any): Pointer
+
+```
+
+From a JavaScript object gets a pointer to the backing native object. The `instance` can be a class constructor (of Objective-C interface), an instance (wrapper of Objective-C interface), struct instance, reference, protocol, function (for c function) or block.
```

Comment:
```suggestion
A JavaScript object can obtain a reference to the underlying native object. The instance can represent a class constructor (for an Objective-C interface), an instance (wrapper of an Objective-C interface), a struct instance, a reference, a protocol, a function (for C functions), or a block.
```

Reply:
really wonderful touches on this doc @SeanKelly369 thank you!